lib/finsta/posts.ex

```diff
@@ -54,9 +54,14 @@ defmodule Finsta.Posts do
     |> Post.changeset(attrs)
     |> Ecto.Changeset.put_assoc(:user, user)
     |> Repo.insert()
+    |> maybe_add_initial_likes()
```

Comment:
to be honest, I don't like this approach...

Reply:
Yes, it does. The thing is we already know the post will have no likes because it was just created, that is why I was manually assigning `[]`, that was something I did not like completely.
Another thing I don't like about the approach is the `maybe` function. Another alternative, relying on `Repo.preload`, can be:

```elixir
 def create_post(user, attrs) do
    %Post{}
    |> Post.changeset(attrs)
    |> Ecto.Changeset.put_assoc(:user, user)
    |> Repo.insert()
    |> case do
      {:ok, post} -> {:ok, Repo.preload(post, [:likes])}
      error -> error
    end
    |> tap(&broadcast_post(&1, :insert))
  end
```

Do you think this approach is better? I generally try to avoid piping with `case` but in this case I think the logic is simple enough to not interfere with readability.

Reply:
I've been discussing this with @augusto1024 and he came up with some interesting ideas

We could replace the `case` with a `then` like this:

```elixir
    ...
    |> Repo.insert()
    |> then(fn {:ok, post} -> {:ok, Repo.preload(post, [:likes])} end)
    |> tap(&broadcast_post(&1, :insert))
```

Or make the component responsible for this
```elixir
  def like(%{post_likes: post_likes} = assigns) when not is_list(post_likes) do
    assigns
    |> Map.put(:post_likes, [])
    |> like()
  end

  def like(assigns) do
      ...
  end
``` 

WDYT? I don't have a strong opinion on either of those

Reply:
Making the component responsible for this is a really great approach! I will implement it that way, thanks both for the help!